File: ios/Sources/Capacitor24hTimePlugin/Capacitor24hTime.swift
import Foundation

@objc public class Capacitor24hTime: NSObject {
    @objc public func echo(_ value: String) -> String {
        print(value)
        return value
    }

     @objc public func is24h() -> Bool {
                  let formatter = DateFormatter()
                  formatter.locale = Locale.current
                  formatter.setLocalizedDateFormatFromTemplate("j")  // 'j' is the time format key

                  // Return true if the device uses 24-hour format, false if it uses 12-hour
                  return !formatter.dateFormat.contains("a")
              }
}

File: android/src/main/java/de/exitare/plugins/capacitor24htime/Capacitor24hTime.java
package de.exitare.plugins.capacitor24htime;

import android.text.format.DateFormat;
import android.content.Context;
import android.util.Log;

public class Capacitor24hTime {

    public String echo(String value) {
        Log.i("Echo", value);
        return value;
    }

        // Method to check if the device is using 24-hour format
           public Boolean is24h(Context context) {
               // Check if the device is using 24-hour format
               return DateFormat.is24HourFormat(context);
           }
}
